core.coffee

```diff
@@ -434,6 +449,57 @@
 						url += '?' + queryOptions.join('&')
 					return url
 
+			compileBatch: (data) ->
+				if !utils.isArray(data)
+					throw new Error('Batch must be passed an array')
+				if !mimemessage?.factory
+					throw new Error('The mimemessage implementation must support .factory')
```

Comment:
Can you move the method checks of `mimemessage` to the same place as the `utils` and `Promise` checks, but only run the checks if `mimemessage` is not null (and change the check here to be that `mimemessage` exists) - that way we do the signature check as soon as `mimemessage` is provided and fail early if it's wrong, but can still have `mimemessage` be optional if a batch request is never attempted